packages/iov-keycontrol/src/userprofile.ts

```diff
@@ -315,21 +315,26 @@ export class UserProfile {
    * creator field specifies the keypair to be used for signing.
    */
   public async signTransaction(
+    identity: Identity,
     transaction: UnsignedTransaction,
     codec: TxCodec,
     nonce: Nonce,
   ): Promise<SignedTransaction> {
-    const wallet = this.findWalletInPrimaryKeyringByIdentity(transaction.creator);
+    const wallet = this.findWalletInPrimaryKeyringByIdentity(identity);
+    const transactionWithExplicitSigner = {
+      ...transaction,
+      creator: identity,
+    };
 
-    const { bytes, prehashType } = codec.bytesToSign(transaction, nonce);
+    const { bytes, prehashType } = codec.bytesToSign(transactionWithExplicitSigner, nonce);
```

Comment:
Can we add `signer: Identity` as a new, second argument to `TxCodec.bytesToSign` and not use the creator anymore in the codec? This would avoid the need for the `transactionWithExplicitSigner` construction

Reply:
I was thinking of doing this in a separate PR, but can definitely include it here.

Reply:
Great, let's merge this then.

Reply:
Actually `bytesToSign` just needs the chain ID, not the full signer anyway.

Reply:
Even better

Reply:
Hmm, looks like we need a sender pubkey for Lisk send transactions:

```ts
      return new Uint8Array([
        0, // transaction type
        ...timestampBytes,
        ...unsigned.creator.pubkey.data,
        ...recipient.toBytesBE(),
        ...amount.toBytesLittleEndian(),
        ...memoBytes,
      ]);
```

Reply:
Ah yeah, I was just thinking about BNS. Hm, wonder why it's the public key not the address?